Source/WebCore/rendering/TextBoxPainter.cpp

```diff
@@ -368,6 +368,20 @@ void TextBoxPainter<TextBoxPath>::paintForegroundAndDecorations()
     } else {
         // Coalesce styles of adjacent marked texts to minimize the number of drawing commands.
         auto coalescedStyledMarkedTexts = StyledMarkedText::coalesceAdjacentWithEqualForeground(styledMarkedTexts);
+        bool isHighlights = false;
+
+        for (auto& element : coalescedStyledMarkedTexts) {
+            if (!isHighlights && element.type == 6) {
+                isHighlights = true;
+                break;
+            }
+        }
+        // Last element is original text with default color, should paint first underneath all marked texts
+        if (isHighlights && coalescedStyledMarkedTexts[coalescedStyledMarkedTexts.size()-1].highlightName.isNull()) {
+            auto originalText = coalescedStyledMarkedTexts[coalescedStyledMarkedTexts.size()-1];
+            coalescedStyledMarkedTexts.remove(coalescedStyledMarkedTexts.size()-1);
+            coalescedStyledMarkedTexts.insert(0, originalText);
+        }
```

Comment:
You could use an init statement here

```suggestion
        if (auto originalText = coalescedStyledMarkedTexts.last(); isHighlights && originalText.highlightName.isNull()) {
            coalescedStyledMarkedTexts.removeLast();
            coalescedStyledMarkedTexts.insert(0, originalText);
        }
```